### DISCUSSION TITLE
Installing and running github-api

### DISCUSSION THREAD

**Discussion by nanyiyang** [hub4j/github-api#1296] [Q&A] [answered]
Sorry if these are dumb questions, but I would appreciate the answer to the following:

1. After installing the jar file from [here](https://mvnrepository.com/artifact/org.kohsuke/github-api/1.135) and importing it into a project, how would I verify that everything is working properly? I tried commands from [here](https://github-api.kohsuke.org/) in a main file, but I was getting ClassNotFoundExeception (see below) and it was saying that the createRepository() command was depreciated. 

```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/io/IOUtils
	at org.kohsuke.github.GitHubBuilder.fromPropertyFile(GitHubBuilder.java:205)
	at org.kohsuke.github.GitHubBuilder.fromPropertyFile(GitHubBuilder.java:186)
	at org.kohsuke.github.GitHubBuilder.fromCredentials(GitHubBuilder.java:70)
	at org.kohsuke.github.GitHub.connect(GitHub.java:261)
	at com.company.Main.main(Main.java:16)

Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.IOUtils
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 5 more
```

2. If I fork and clone the repo, is there a way to import this repo and use it like the library from the jar file?

Once again sorry if these are dumb questions, I'm quite new to Java development. Thanks!

**Comment by bitwiseman** [comment, marked as answer]
> Sorry if these are dumb questions, but I would appreciate the answer to the following:
> 
> 1. After installing the jar file from [here](https://mvnrepository.com/artifact/org.kohsuke/github-api/1.135) and importing it into a project, how would I verify that everything is working properly? I tried commands from [here](https://github-api.kohsuke.org/) in a main file, but I was getting ClassNotFoundExeception (see below) and it was saying that the createRepository() command was depreciated.
> 
> ```
> Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/io/IOUtils
> 	at org.kohsuke.github.GitHubBuilder.fromPropertyFile(GitHubBuilder.java:205)
> 	at org.kohsuke.github.GitHubBuilder.fromPropertyFile(GitHubBuilder.java:186)
> 	at org.kohsuke.github.GitHubBuilder.fromCredentials(GitHubBuilder.java:70)
> 	at org.kohsuke.github.GitHub.connect(GitHub.java:261)
> 	at com.company.Main.main(Main.java:16)
> 
> Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.IOUtils
> 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
> 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
> 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
> 	... 5 more
> ```

Use maven and a pom.xml to build your project.  This will pull in all your dependencies and the things they depend on (in this case Apache Commons IO. 

The deprecated message tells you what to do but we should also update that example to use the newer method:

https://github.com/hub4j/github-api/blob/1535817992bec2a594237532ab9a6e05542f0e8a/src/main/java/org/kohsuke/github/GitHub.java#L962


> 1. If I fork and clone the repo, is there a way to import this repo and use it like the library from the jar file?

In java, no I don't think so.  Sorry. 

> Once again sorry if these are dumb questions, I'm quite new to Java development. Thanks!

No problem.  There's a bit of learning curve.
